pkg/security/secl/rules/ruleset.go

```diff
@@ -487,7 +488,9 @@ func (rs *RuleSet) NotifyRuleMatch(rule *Rule, event eval.Event) {
 	defer rs.listenersLock.RUnlock()
 
 	for _, listener := range rs.listeners {
-		listener.RuleMatch(rule, event)
+		if !listener.RuleMatch(rule, event) {
```

Comment:
why do we need this change ?

Reply:
The reason is I wanted to decouple the self tester and the rule engine. The reason is that the self tester was called in the `RuleMatch` method, to decide whether the event should be sent.
So I made the self tester and new listener and also allowed a listener to ask the other listeners to not be called.

Does it make sense ?

Reply:
yes makes sense, thanks